src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,719 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include "info_aie.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace qr = xrt_core::query;
+using ptree_type = boost::property_tree::ptree;
+
+#define MAJOR_VERSION_NUMBER 1
+#define MINOR_VERSION_NUMBER 0
+#define PATCH_VERSION_NUMBER 0
+
+enum class graph_state {
+  STOP		= 0,
+  RESET		= 1,
+  RUNNING	= 2,
+  SUSPEND	= 3,
+  END		= 4
+};
+
+namespace {
+inline std::string
+graph_status_to_string(int status) {
+  switch(static_cast<graph_state>(status))
+  {
+    case graph_state::STOP:	return "stop";
+    case graph_state::RESET:	return "reset";
+    case graph_state::RUNNING:	return "running";
+    case graph_state::SUSPEND:	return "suspend";
+    case graph_state::END:	return "end";
+    default:			return "idle";
```

Comment:
Just looking at the code more carefully, what is this state "idle"?
Does this means that the `enum class graph_state` can have a value outside of the defined `enum` values???
While this is definitely possible, it is very non advisable.

Reply:
There are a few problems with this code.
(1) The enum class is useless, it doesn't communicate anything.  The values comes in as int and are cast to graph_state but graph state serves no other purpose.   It's just not cool :-), so get rid of graph_state or use it all over the place, which is difficult since the state comes in as an int from a property tree.
(2) Never use UPPPER NAME enumerators in class enums, and especially never UPPERCASE generic enumerators like STOP, END, etc. as are being used here.    The preprocessor can be affected by someone #defining these UPPERCASE enumerators.

Reply:
Ok. Removed.